cpp/ql/src/Security/CWE/CWE-129/ImproperArrayIndexValidation.ql

```diff
@@ -66,7 +66,10 @@ predicate predictableInstruction(Instruction instr) {
 }
 
 module ImproperArrayIndexValidationConfig implements DataFlow::ConfigSig {
-  predicate isSource(DataFlow::Node source) { isFlowSource(source, _) }
+  predicate isSource(DataFlow::Node source) {
+    isFlowSource(source, _) and
+    not source.getLocation().getFile().getRelativePath().regexpMatch("(.*/)?tests?/.*")
```

Comment:
Do we not have other ways of doing this (like some kind of filtering from Liquid post sarif generation)? If not, that's kind of crazy because we would need to add something like this to all of our queries, right? Or am I missing something

Reply:
Yes, and usually we wouldn't make such a change and just use the post-analysis finding filtering, however if a CodeQL rule generates over 2GB of data for an individual rule, the CLI will throw an exception and not complete and we are seeing instances of this. More detail in DM.